infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
What makes you believe that we can always factor out a monadic-context-free `MyData` piece?

We're hoping to wire up @edsko 's relayer soon. This one takes care of queueing, and of a thread to dequeue and update internal state, but it doesn't specify how messages are sent. Instead, it takes a callback `PropagationMsg -> NodeId -> m ()`. Notice the `m`. It also has configuration, some of which also mentions the application monad `m`: `rcfgDelay :: NodeId -> m (Maybe Microsecond)` to compute a delay for the next send to this node, `rcfgNextSlotLeader :: m (Maybe NodeId)` to compute the next slot leader (if known). This relayer is nowhere near as simple as the original implementation (a `TBQueue`). Its internal workings depend upon the application monad: we can't pull out a simple `MyData` from such a relayer. It really truly does depend upon the ultimate application monad.